### DISCUSSION TITLE
Help to use excalidraw with html/js only

### DISCUSSION THREAD

**Discussion by gitubbi** [excalidraw/excalidraw#6537] [Q&A]
Hello,

Sorry for the naivety of my question.

I am trying to use Excalidraw in a self-hosted local site, but i don't know React. 

So I have followed the doc to integrate Excalidraw in a simple html file :

```
    <div class="container">
       <div id="excali"></div>
    </div>

    <script> 
    
       const app = () => {
           react = React.createElement( React.Fragment, null,
                      React.createElement( "div", { style: { height: "100vh" }, },
                      React.createElement( ExcalidrawLib.Excalidraw ),
                   ),);

       const excali = document.getElementById("excali");
       const root   = ReactDOM.createRoot(excali);
             root.render(React.createElement(app));
    
    </script>
```

It works well and i can use excalidraw without problem. But now i wish to add some library files at start (.excalidrawlib). 

The library file is on server side and is asked by client side with fetch. The library file is well received as a json file.

But then I don't know how to update the library with this content.

I have tried to use :

`ExcalidrawLib.loadLibraryFromBlob()  or excalidrawAPI.updateLibrary() `

without success, because the examples in doc are with JSX syntax that I don't know how to use.

As far as i have understood you have to create 2 constants called hooks:

`const [ excalidrawAPI, setexcalidrawAPI] = React.useState(null);`
   
Then init excalidrawAPI with:   `setexcalidrawAPI(api)`
but i don't know where to get api object

and then update library with :
   
     `excalidrawAPI.updateLibrary(libraryItems)`

But as i'm stucked at step: `setexcalidrawAPI(api)`, is there a way to update the library only with html/js syntax (not react) ?

Thank you very much for any help

**Comment by gitubbi** [comment]
I answer to my own question, if it can helps someone one day:

You can pass library and element at load with React.createElement, if you add initialData in options :

```
    <div class="container">
      <div id="excali"></div>
    </div>
    
    <script>
    
         var excalidrawAPI, setExcalidrawAPI;

         var options =  {     
                langCode: 'fr-FR',
                ref: (api) => setExcalidrawAPI(api), 
                initialData: {
                   libraryItems: libraryFile.libraryItems,   //from a libray file
                   elements:   [ someelement ]
                }
             };
                  
         const App = () => {

             [excalidrawAPI, setExcalidrawAPI] = React.useState(null);
  
             return React.createElement( React.Fragment, null,
                           React.createElement( "div", { style: { height: "100vh" }, },
                           React.createElement( ExcalidrawLib.Excalidraw, options ), ),  //put options here
                    );
         };  

       const excali = document.getElementById("excali");
       const root   = ReactDOM.createRoot(excali);
             root.render(React.createElement(App));
             
    </script>
```

**Reply by ad1992** [reply]
Hey @gitubbi glad to know you figured it out. You can refer to the  [docs](https://docs.excalidraw.com/docs/@excalidraw/excalidraw/api/props/initialdata) for more info and examples. The API remains the same whether you using directly in browser or with bundler.

**Reply by GurdeepSingh55** [reply]
Hi @gitubbi @ad1992 

Is there any way to get the selected element using excalidraw API?

**Reply by ad1992** [reply]
you can get the [appState](https://docs.excalidraw.com/docs/@excalidraw/excalidraw/api/props/excalidraw-api#getappstate) and in `appState` you can check `selectedElementIds`, that should give you the elements which are selected.

**Comment by messi20221218** [comment]
thx, it help me a lot!
